src/Mesh/Gradients.C

```diff
@@ -0,0 +1,165 @@
+// *****************************************************************************
+/*!
+  \file      src/Mesh/Gradients.C
+  \copyright 2012-2015, J. Bakosi, 2016-2017, Los Alamos National Security, LLC.
+  \brief     Functions computing gradients on unstructured meshes for tetrahedra
+  \details   Functions computing gradients using linear finite element shape
+             functions on unstructured meshes for tetrahedra.
+*/
+// *****************************************************************************
+
+#include "Exception.h"
+#include "Gradients.h"
+#include "Vector.h"
+
+namespace tk {
+
+std::array< tk::real, 3 >
+nodegrad( std::size_t node,
+          const std::array< std::vector< tk::real >, 3 >& coord,
+          const std::vector< std::size_t >& inpoel,
+          const std::pair< std::vector< std::size_t >,
+                           std::vector< std::size_t > >& esup,
+          const tk::Fields& U,
+          ncomp_t c )
+// *****************************************************************************
+//  Compute gradient at a mesh node
```

Comment:
Should we also settle on a style of consistent comments? I feel like most files have the comment before the method signature (my preferred way, it can also be consistent with headers then), but some have it between the definition and the body

Reply:
In headers doxygen wants the comments before and in the .C files we have the comments section after the signature and before the definition. This is consistent with that right?